pennylane/_qubit_device.py

```diff
@@ -779,9 +785,13 @@ def var(self, observable, shot_range=None, bin_size=None):
             )
             return probs[idx] - probs[idx] ** 2
 
+        # exact variance value
         if self.shots is None:
-            # exact variance value
-            eigvals = self._asarray(observable.eigvals, dtype=self.R_DTYPE)
+            try:
+                eigvals = self._asarray(observable.eigvals, dtype=self.R_DTYPE)
+            except NotImplementedError as e:
+                # if observable has no info on eigenvalues, we cannot return this measurement
+                raise ValueError(f"Cannot compute samples of {observable.name}.") from e
```

Comment:
```suggestion
                raise ValueError(f"Cannot compute analytic variance of {observable.name}.") from e
```
Based on the comment heading in this section?

Reply:
Ah good catch!